Assets/Scripts/UI/LogText/TurnEnd.cs

```diff
@@ -16,7 +16,58 @@ public override IEnumerable<LogRecord> Convert(Ffb.Dto.Reports.TurnEnd report)
             {
                 yield return new LogRecord($"{scorer.FormattedName} scores a touchdown.", 1);
             }
-            // TODO: unfinished
+            if (0 < report.knockoutRecoveryArray?.Length)
+            {
+                string s = String.Empty;
+                foreach (Fumbbl.Ffb.Dto.Reports.KnockoutRecoveryResult res in report.knockoutRecoveryArray)
+                {
+                    Player player = FFB.Instance.Model.GetPlayer(res.playerId);
+
+                    s = $"<b>Knockout Recovery Roll [ {res.roll} ]</b>";
+                    if (0 < res.bloodweiserBabes)
+                    {
+                        s += $"<b> + {res.bloodweiserBabes} Bloodweiser Babes</b>";
+                    }
+                    yield return new LogRecord(s);
+                    if (res.recovering)
+                    {
+                        yield return new LogRecord($"{player.FormattedName} is regaining consciousness.", 1);
+                    }
+                    else
+                    {
+                        yield return new LogRecord($"{player.FormattedName} stays unconscious.", 1);
+                    }
+                }
+            }
+            if (0 < report.heatExhaustionArray?.Length)
+            {
+                string s = String.Empty;
+                foreach (Fumbbl.Ffb.Dto.Reports.HeatExhaustionResult res in report.heatExhaustionArray)
+                {
+                    Player player = FFB.Instance.Model.GetPlayer(res.playerId);
+
+                    s = $"<b>Heat Exhaustion Roll [ {res.roll} ]</b>";
+                    if (res.exhausted)
+                    {
+                        yield return new LogRecord($"{player.FormattedName} is suffering from heat exhaustion.", 1);
+                    }
+                    else
+                    {
+                        yield return new LogRecord($"{player.FormattedName} is unaffected.", 1);
+                    }
+                }
+            }
+            if (FFB.Instance.Model.TurnMode == TurnMode.Regular)
+            {
+                if (FFB.Instance.Model.HomePlaying)
+                {
+                    yield return new LogRecord($"<size=16><color=#ff0000>{FFB.Instance.Model.TeamHome.Name} start turn {FFB.Instance.Model.TurnHome}.</color></size>");
```

Comment:
Here's a typical example of re-use of the team colours. Makes sense to have this as a constant somewhere suitable.